File: go/streaming/internal/stream_asr_job_outgoing_message/close_stream_asr_job_message.go
package stream_asr_job_outgoing_message

type CloseStreamAsrJobMessage struct {
	Command string `json:"command"`
}

func NewCloseStreamAsrJobMessage() CloseStreamAsrJobMessage {
	return CloseStreamAsrJobMessage{
		Command: "close_stream_asr_job",
	}
}

File: cpp/src/poetics_standalone/poetics_sdk/streaming/asr_job/outgoing_message/close_stream_asr_job_message.cpp
#include "close_stream_asr_job_message.h"

namespace poetics::streaming::asr_job::outgoing_message {
    CloseStreamAsrJobMessage::CloseStreamAsrJobMessage(string _command) : command(_command) {}

    CloseStreamAsrJobMessage CloseStreamAsrJobMessage::CreateDefault() { 
        return CloseStreamAsrJobMessage("close_stream_asr_job"); 
    }

    void to_json(json& j, const CloseStreamAsrJobMessage& message) {
        j = json{{"command", message.command}};
    }
}
